src/ert/ensemble_evaluator/_builder/_legacy.py

```diff
@@ -120,6 +120,30 @@ def evaluate(self, config: EvaluatorServerConfig) -> None:
 
         ErtThread(target=self._evaluate, name="LegacyEnsemble").start()
 
+    async def evaluate_async(self, config: EvaluatorServerConfig) -> None:
+        if not config:
+            raise ValueError("no config for evaluator")
+        self._config = config
+        await wait_for_evaluator(
+            base_url=self._config.url,
+            token=self._config.token,
+            cert=self._config.cert,
+        )
+        ce_unary_send_method_name = "_ce_unary_send"
+        setattr(
+            self.__class__,
+            ce_unary_send_method_name,
+            partialmethod(
+                self.__class__.send_cloudevent,
+                self._config.dispatch_uri,
+                token=self._config.token,
+                cert=self._config.cert,
+            ),
+        )
+        await self._evaluate_inner(
+            cloudevent_unary_send=getattr(self, ce_unary_send_method_name)
+        )
```

Comment:
Can all of these lines be replaced with this?
```
await self._evaluate_inner(
   cloudevent_unary_send=partialmethod(
      self.__class__.send_cloudevent,
      self._config.dispatch_uri,
      token=self._config.token,
      cert=self._config.cert,
   )
)
```

Reply:
Unfortunately it din't work so need to keep it as is.

Reply:
Why did it not work?
There is probably a reason why they made it this way, but I am curious to know why.

Reply:
It complained (when starting the ee loop) that there is no running loop.

Reply:
The signature of the function changed as if the partialmehod does not preserve the async property. In this case though we attach partial to a class member, which does some magic apparently.